roda-core/roda-core/src/main/java/org/roda/core/RodaCoreFactory.java

```diff
@@ -2192,10 +2192,12 @@ private static String readPassword(final String message) throws IOException {
 
   private static void resetAdminAccess() throws GenericException {
     try {
-      final String password = readPassword("New admin password");
-      final String passwordConfirmation = readPassword("Repeat admin password");
-      if (password.equals(passwordConfirmation)) {
+      char[] password = readPassword("New admin password").toCharArray();
+      char[] passwordConfirmation = readPassword("Repeat admin password").toCharArray();
+      if (Arrays.equals(password, passwordConfirmation)) {
         RodaCoreFactory.ldapUtility.resetAdminAccess(password);
+        password = null;
+        passwordConfirmation = null;
```

Comment:
modify the readPassword method to return SecureString object and use "try-with-resouce"